http/src/main/java/io/micronaut/http/hateos/DefaultLink.java

```diff
@@ -22,113 +22,20 @@
 import java.util.Optional;
 
 /**
- * Default implementation of {@link Link}.
+ * Deprecated. Please use io.micronaut.http.hateoas.DefaultLink
  *
  * @author Graeme Rocher
  * @since 1.0
  */
-class DefaultLink implements Link, Link.Builder {
-
-    final URI href;
-    private boolean templated;
-    private URI profile;
-    private URI deprecation;
-    private String title;
-    private String hreflang;
-    private MediaType type;
-    private String name;
+@Deprecated
+class DefaultLink extends io.micronaut.http.hateoas.DefaultLink {
 
     /**
      * @param uri The URI
      */
+    //TODO AGB don't know how to solve this
```

Comment:
don't know how to solve this: protected class (not visible from another package) which should remain like that.
But the packages should be different (hateoas vs hateos). 
What do you think?